overlord/hookstate/hooks.go

```diff
@@ -77,6 +78,26 @@ func SetupPreRefreshHook(st *state.State, snapName string) *state.Task {
 	return task
 }
 
+// AffectingSnapsForAffectedByRefreshCandidates returns the list of all snaps
+// affecting affectedSnap (i.e. a gating snap), based on upcoming updates
+// from refresh-candidates.
+func AffectingSnapsForAffectedByRefreshCandidates(st *state.State, affectedSnap string) ([]string, error) {
```

Comment:
doesn't this fits better in snapstate?

Reply:
I was considering it but felt it was very specialized for the hooks here and not useful elsewhere. I can move it if you prefer it in snapstate.

Reply:
the problem is that  needs to be exported and reads strangely as part of the rest of the hookstate API, so yes I think if it must be exported it fits better over there

Reply:
Fair point, will move it.

Reply:
Done.